packages/autorest.typescript/src/utils/schemaHelpers.ts

```diff
@@ -258,13 +258,21 @@ export function getSecurityInfoFromModel(security: Security) {
       );
     }
   }
-  const scopes: string[] = [];
-  credentialScopes.forEach(item => {
-    scopes.push(item);
-  });
+  const scopes: string[] = [...credentialScopes];
+  let refinedAddCredentials =
+    addCredentials === false ? false : security.authenticationRequired;
+  // Add default info if explictly set the addCredentials as true
+  if (
```

Comment:
Set the default value as token credential with scope `https://management.azure.com/.default`.